### PR TITLE
Documentation: Update list of available shell

### PR DESCRIPTION
**Pull request by Emplis**
- Remove OS specific instructions, instructions depend on the shell and are the same across OS;
- Add fish shell to the documentation.

Online Documentation build: https://builds.zephyrproject.io/zephyr/pr/62785/docs/develop/west/install.html

### CODE REVIEW

**Comment by Emplis** [comment]
@kartben I added documentation but only for Linux as I don't have macOS or Windows computers.

I suppose that the Zsh completions can be used without any issues on macOS as it's the default shell now but I can't test it.

File: doc/develop/west/install.rst [hunk 1]
```diff
@@ -51,7 +51,7 @@ Enabling shell completion
 West currently supports shell completion in the following combinations of
 platform and shell:
 
-* Linux: bash
+* Linux: bash, zsh, fish
 * macOS: bash
 * Windows: not available
```
Comment (nguyenmthien) [inline doc/develop/west/install.rst:56]: Can we remove this whole section? One can get Bash on Windows with MinGW32

File: doc/develop/west/install.rst [hunk 2]
```diff
@@ -72,11 +72,31 @@ Next, you need to import :file:`west-completion.bash` into your bash shell.
...
+
+  .. group-tab:: fish
+
+    Run the following command.
+
+    .. code-block:: fish
+
+      west completion fish > $HOME/.config/fish/completions/west.fish
 
 On macOS, you have the following options:
```
Comment (nguyenmthien) [inline doc/develop/west/install.rst:101]: Similiarly, the setup really depends on the shell and not the OS. Can we remove the OS-specific section here?

File: doc/develop/west/install.rst
Comment (nguyenmthien) [inline doc/develop/west/install.rst:1]: Can we also change line 42 of the file `doc/develop/west/zephyr-cmds.rst` to add fish shell there?

**Review by carlescufi — DISMISSED** [review]
Thanks!

File: doc/develop/west/install.rst (hunk 1 above)
Comment (carlescufi) [reply doc/develop/west/install.rst:56]: Yes, that is true. I agree with @nguyenmthien.

File: doc/develop/west/install.rst
Comment (carlescufi) [reply doc/develop/west/install.rst:1]: Yes please.

**Review by mbolivar-ampere — COMMENTED** [review]
Agree with @nguyenmthien

**Comment by Emplis** [comment]
@carlescufi @nguyenmthien @mbolivar-ampere
Sorry I forgot about this PR. I updated it following the request of @nguyenmthien.

**Comment by MaureenHelm** [comment]
@carlescufi @mbolivar-ampere ping

File: doc/develop/west/install.rst [hunk 3]
```diff
@@ -48,50 +48,59 @@ APIs directly. See :ref:`west-apis` for details.
...
+
+    .. code-block:: bash
+
+      source <(west completion bash)
+
+    *Permanent setup*:
+
+    .. code-block:: bash
+
+      west completion bash > ~/west-completion.bash
```
Comment (kartben) [inline doc/develop/west/install.rst:75]: is this really enough? the initial instructions mentioned that this file must be created *AND* sourced form .bashrc? Is this automatically done based on the name of the file? or you simply forgot it?

Comment (Emplis) [reply doc/develop/west/install.rst:75]: You're right, it's not enough. There is multiple options to edit the line:
1. `west completion bash | sudo tee /usr/share/bash-completion/completions/west-completion.bash` 
2. `west completion bash | sudo tee /etc/bash_completion.d/west-completion.bash`
3. `west completion bash > ~/west-completion.bash; echo "source ~/west-completion.bash" >> ~/.bashrc`

IMO the 1 is the most okay-ish. But maybe without the `sudo` for both 1 and 2.
Do you have an opinion on which one we should choose?

Comment (Emplis) [reply doc/develop/west/install.rst]: Updated the PR with option 3 as it's the most compatible (work even if bash-completion not installed as far as I understand).

**Review by nguyenmthien — APPROVED** [review]
LGTM :smiley:

**Review by jori-nordic — APPROVED** [review]

**Comment by jori-nordic** [comment]
@kartben It's been a few months, can we merge this?

**Review by kartben — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**